exporter/lokiexporter/factory_next.go

```diff
@@ -0,0 +1,39 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lokiexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/lokiexporter"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+func createNextLogsExporter(ctx context.Context, set component.ExporterCreateSettings, cfg *Config) (component.LogsExporter, error) {
```

Comment:
I would keep the name "createLogsExporter" and move it to factory.go.

Reply:
The default route is still to use the "legacy", which is the reason I chose "next" as the name for this. I'll eventually change the code path to use this by default and rename it to reflect the change.